File: spec/javascript/test_bits_byte_aligned.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('BitsByteAligned', 'src/fixed_struct.bin', function(r) {
    // 50 (6 + 2) = 010100|00
    assert.equal(r.one, 0b010100);
    // 41
    assert.equal(r.byte1, 0x41);
    // 43 (3 + 1 + 4) = 010|0|0011
    assert.equal(r.two, 0b010);
    assert.equal(r.three, false);
    // 4B
    assert.equal(r.byte2, 0x4b);
    // 2D 31 (14 + 2) = 00101101 001100|01
    assert.equal(r.four, 0b00101101001100);
    // FF
    assert.equal(r.byte3.toString(), new Uint8Array([0xff]).toString());
    // FF
    assert.equal(r.fullByte, 0xff);
    // 50
    assert.equal(r.byte4, 0x50);
});

File: spec/lua/test_bits_byte_aligned.lua
local luaunit = require("luaunit")

require("bits_byte_aligned")

TestBitsByteAligned = {}

function TestBitsByteAligned:test_bits_byte_aligned()
    local r = BitsByteAligned:from_file("src/fixed_struct.bin")

    -- 50 (6 + 2) = 010100|00
    luaunit.assertEquals(r.one, 0x14)
    -- 41
    luaunit.assertEquals(r.byte_1, 0x41)
    -- 43 (3 + 1 + 4) = 010|0|0011
    luaunit.assertEquals(r.two, 0x02)
    luaunit.assertEquals(r.three, 0)
    -- 4B
    luaunit.assertEquals(r.byte_2, 0x4b)
    -- 2D 31 (14 + 2) = 00101101 001100|01
    luaunit.assertEquals(r.four, 0xb4c)
    -- FF
    luaunit.assertEquals(r.byte_3, "\255")
    -- FF
    luaunit.assertEquals(r.full_byte, 0xff)
    -- 50
    luaunit.assertEquals(r.byte_4, 0x50)
end
